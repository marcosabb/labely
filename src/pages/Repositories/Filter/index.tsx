import React, { useCallback, useState } from 'react'
import { useDebouncedCallback } from 'use-debounce'

import { useRepositories } from 'contexts/repositories'
import { useUsers } from 'contexts/users'

import Input from 'components/Input'

import {
  Container,
  Fields,
  Labels,
  LabelItem,
  LabelText,
  CreateButton,
  PlusIcon,
  DeleteButton,
  CloseIcon
} from './styles'

interface Filters {
  name: string
  labels: string[]
}

interface Expanded {
  name: boolean
  label: boolean
}

export default function Filter() {
  const { filterRepositories } = useRepositories()
  const { currentUser } = useUsers()

  const [label, setLabel] = useState('')
  const [filters, setFilters] = useState<Filters>({
    name: '',
    labels: []
  })
  const [expanded, setExpanded] = useState<Expanded>({
    name: true,
    label: false
  })

  const handleChangeLabel = useCallback((value) => {
    setLabel(value)
  }, [])

  const handleCreateLabel = useCallback(() => {
    setLabel('')
    setFilters((state) => ({
      ...state,
      labels: [...new Set([...state.labels, label])]
    }))
  }, [label])

  const handleRemoveLabel = useCallback((label) => {
    setFilters((state) => ({
      ...state,
      labels: state.labels.filter((_, index) => index !== label)
    }))
  }, [])

  const handleChangeName = useCallback(async (value) => {
    setFilters((state) => ({
      ...state,
      name: value
    }))
  }, [])

  const debounced = useDebouncedCallback(async () => {
    await filterRepositories(currentUser, filters)
  }, 1000)

  const handleExpand = useCallback(
    (key: string) => {
      const filterState = Object.keys(filters).reduce(
        (total, current) => ({
          ...total,
          [current]: Array.isArray(current) ? [] : ''
        }),
        {}
      ) as Filters

      const expandedState = Object.keys(expanded).reduce(
        (total, current) => ({ ...total, [current]: key === current }),
        {}
      ) as Expanded

      setFilters(filterState)
      setExpanded(expandedState)
    },
    [expanded, filters]
  )

  return (
    <Container>
      <Fields>
        <Input
          name='name'
          value={filters.name}
          icon='search'
          color='primary'
          onChangeText={(value) => {
            handleChangeName(value)
            debounced.callback()
          }}
          onExpand={() => {
            !expanded.name && handleExpand('name')
            debounced.callback()
          }}
          expanded={expanded.name}
          direction={expanded.name ? 'right' : undefined}
        />

        <Input
          name='label'
          value={label}
          icon='filter-list'
          color='primary'
          onChangeText={handleChangeLabel}
          onExpand={() => {
            !expanded.label && handleExpand('label')
            debounced.callback()
          }}
          expanded={expanded.label}
          direction={expanded.label ? 'left' : undefined}
        />
      </Fields>

      <Labels>
        {!!filters.labels &&
          filters.labels.length > 0 &&
          filters.labels.map((label, index) => (
            <LabelItem key={label}>
              <LabelText numberOfLines={1}>{label}</LabelText>

              <DeleteButton
                onPress={() => {
                  handleRemoveLabel(index)
                  debounced.callback()
                }}
              >
                <CloseIcon name='close' />
              </DeleteButton>
            </LabelItem>
          ))}

        {!!label && (
          <LabelItem>
            <LabelText numberOfLines={1}>{label}</LabelText>

            <CreateButton
              onPress={() => {
                handleCreateLabel()
                debounced.callback()
              }}
            >
              <PlusIcon name='add' />
            </CreateButton>
          </LabelItem>
        )}
      </Labels>
    </Container>
  )
}
