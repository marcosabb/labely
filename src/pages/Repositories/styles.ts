import styled from 'styled-components/native'
import { theme } from 'styled-tools'
import { FlatList } from 'react-native'

import { Repository } from '../../contexts/repositories'

export const Container = styled.View`
  flex: 1;
  padding-top: ${theme('spacing.xs')};
  background-color: ${theme('colors.background')};
`

export const HeaderRight = styled.View`
  margin-right: ${theme('spacing.default')};
`

export const Avatar = styled.Image`
  width: 32px;
  height: 32px;
  border-radius: 16px;
`

export const List = styled(FlatList as new () => FlatList<Repository>).attrs({
  contentContainerStyle: {
    paddingTop: 8,
    paddingHorizontal: 16
  }
})``

export const Loading = styled.ActivityIndicator.attrs(({ theme }) => ({
  color: theme.colors.black
}))`
  flex: 1;
`
