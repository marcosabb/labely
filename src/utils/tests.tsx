import React from 'react'
import { render, RenderOptions } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'

import theme from 'styles/theme'

interface Props {
  children: React.ReactNode
}

function ThemeWrapper({ children }: Props) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: ThemeWrapper, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
