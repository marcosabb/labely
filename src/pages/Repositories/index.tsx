import React, { useCallback, useEffect } from 'react'
import { useNavigation, useRoute, RouteProp } from '@react-navigation/native'
import { formatDistance, parseISO } from 'date-fns'
import ptBR from 'date-fns/locale/pt-BR'

import { Repository, useRepositories } from 'contexts/repositories'

import { RootStackParamList } from 'routes'

import Item from 'components/Item'
import Filter from './Filter'

import { Container, List, HeaderRight, Avatar, Loading } from './styles'

interface RenderProps {
  item: Repository
}

export default function Repositories() {
  const { navigate, setOptions } = useNavigation()
  const { params } = useRoute<RouteProp<RootStackParamList, 'Repositories'>>()

  const {
    repositories,
    loading,
    getRepositories,
    setSelectedRepository
  } = useRepositories()

  useEffect(() => {
    async function fetchRepositories() {
      const { user } = params
      const { login } = user

      await getRepositories(login)
    }

    fetchRepositories()
  }, [getRepositories, params])

  useEffect(() => {
    const { user } = params

    setOptions({
      headerRight: () => (
        <HeaderRight>
          <Avatar source={{ uri: user.avatar_url }} />
        </HeaderRight>
      )
    })
  }, [params, setOptions])

  const handleNavigateToActions = useCallback(
    (repository) => {
      navigate('Actions')
      setSelectedRepository(repository)
    },
    [navigate, setSelectedRepository]
  )

  function keyExtractor({ id }: Repository) {
    return String(id)
  }

  function renderItem({ item }: RenderProps) {
    const {
      name,
      description,
      language,
      stargazers_count,
      updated_at,
      labels
    } = item

    const tags = [
      { id: 1, icon: 'language', value: language },
      { id: 2, icon: 'star', value: stargazers_count },
      {
        id: 3,
        icon: 'query-builder',
        value: formatDistance(parseISO(updated_at), new Date(), {
          locale: ptBR
        })
      }
    ]

    return (
      <Item
        title={name}
        description={description}
        labels={labels}
        tags={tags}
        onPress={() => handleNavigateToActions(item)}
        highlight
      />
    )
  }

  return (
    <Container>
      <Filter />

      {loading.repositories ? (
        <Loading size='large' />
      ) : (
        <List
          data={repositories}
          keyExtractor={keyExtractor}
          renderItem={renderItem}
        />
      )}
    </Container>
  )
}
