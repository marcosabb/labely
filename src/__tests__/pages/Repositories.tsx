import React from 'react'
import { fireEvent } from '@testing-library/react-native'

import Repositories from 'pages/Repositories'
import { render } from 'utils/tests'

const mockedNavigate = jest.fn()
const mockedRepositories = [
  {
    id: 76504246,
    login: 'marcosabb',
    name: 'metro',
    description: 'ðŸš‡ The JavaScript bundler for React Native.',
    labels: [],
    language: 'JavaScript',
    stargazers_count: 3956,
    updated_at: '2021-02-21T11:57:39Z'
  },

  {
    id: 41523595,
    login: 'marcosabb',
    name: 'react-joyride',
    description: 'Create guided tours in your apps',
    labels: [],
    language: 'JavaScript',
    stargazers_count: 3554,
    updated_at: '2021-02-21T22:42:34Z'
  }
]

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockedNavigate,
    setOptions: () => ({
      headerLeft: () => {},
      headerRight: () => {}
    })
  }),
  useRoute: () => ({
    params: {
      user: {}
    }
  })
}))

jest.mock('../../contexts/repositories', () => ({
  useRepositories: () => ({
    repositories: mockedRepositories,
    loading: {},
    setSelectedRepository: jest.fn()
  })
}))

describe('Repositories page', () => {
  beforeEach(() => {
    mockedNavigate.mockClear()
  })

  it('should match snapshot', () => {
    const component = render(<Repositories />)

    expect(component).toMatchSnapshot()
  })

  it('should render items if has repositories', () => {
    const { getByText } = render(<Repositories />)

    const name = getByText('react-joyride')

    expect(name).toBeTruthy()
  })

  it('should be able to navigate to actions page when click item', () => {
    const { getByText } = render(<Repositories />)

    const item = getByText('metro')

    fireEvent.press(item)

    expect(mockedNavigate).toHaveBeenCalledWith('Actions')
  })
})
