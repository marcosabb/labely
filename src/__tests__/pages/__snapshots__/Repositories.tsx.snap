// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories page should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 8,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginBottom": 16,
          },
        ]
      }
    >
      <View
        accessible={true}
        direction="right"
        expanded={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#e5e5e5",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "flexGrow": 1,
              "height": 40,
              "marginRight": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "primary",
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#7e7e7e",
                  "fontSize": 24,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          name="name"
          onChangeText={[Function]}
          placeholderTextColor="#7e7e7e"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "#000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Mulish-Regular",
                "fontSize": 16,
                "height": 40,
                "marginLeft": 8,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        expanded={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#e5e5e5",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "width": 48,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "primary",
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#7e7e7e",
                  "fontSize": 24,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          name="label"
          onChangeText={[Function]}
          placeholderTextColor="#7e7e7e"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "#000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Mulish-Regular",
                "fontSize": 16,
                "height": 40,
                "marginLeft": 8,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 8,
      }
    }
    data={
      Array [
        Object {
          "description": "🚇 The JavaScript bundler for React Native.",
          "id": 76504246,
          "labels": Array [],
          "language": "JavaScript",
          "login": "marcosabb",
          "name": "metro",
          "stargazers_count": 3956,
          "updated_at": "2021-02-21T11:57:39Z",
        },
        Object {
          "description": "Create guided tours in your apps",
          "id": 41523595,
          "labels": Array [],
          "language": "JavaScript",
          "login": "marcosabb",
          "name": "react-joyride",
          "stargazers_count": 3554,
          "updated_at": "2021-02-21T22:42:34Z",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {},
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 8,
              "marginBottom": 8,
              "opacity": 1,
              "paddingBottom": 32,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 32,
            }
          }
          testID="item-container"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginBottom": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "Mulish-Bold",
                        "fontSize": 16,
                        "marginBottom": 10,
                        "marginRight": 10,
                        "maxWidth": "50%",
                      },
                    ]
                  }
                  testID="item-title"
                >
                  metro
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontSize": 24,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(255,199,0,0.16)",
                        "borderRadius": 12,
                        "height": 24,
                        "justifyContent": "center",
                        "width": 24,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#ffc700",
                            "fontSize": 16,
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                🚇 The JavaScript bundler for React Native.
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "marginBottom": 16,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0017ff",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "marginTop": 2,
                  "opacity": 1,
                  "width": 16,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 10,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                JavaScript
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                3956
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                cerca de 15 horas
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 8,
              "marginBottom": 8,
              "opacity": 1,
              "paddingBottom": 32,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 32,
            }
          }
          testID="item-container"
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginBottom": 16,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "Mulish-Bold",
                        "fontSize": 16,
                        "marginBottom": 10,
                        "marginRight": 10,
                        "maxWidth": "50%",
                      },
                    ]
                  }
                  testID="item-title"
                >
                  react-joyride
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontSize": 24,
                        },
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(255,199,0,0.16)",
                        "borderRadius": 12,
                        "height": 24,
                        "justifyContent": "center",
                        "width": 24,
                      },
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#ffc700",
                            "fontSize": 16,
                          },
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Create guided tours in your apps
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "marginBottom": 16,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0017ff",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "marginTop": 2,
                  "opacity": 1,
                  "width": 16,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 10,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                JavaScript
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                3554
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginRight": 16,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#e5e5e5",
                        "fontSize": 16,
                        "marginRight": 4,
                      },
                    ],
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7e7e7e",
                      "fontFamily": "Mulish-Regular",
                      "fontSize": 12,
                    },
                  ]
                }
              >
                cerca de 5 horas
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
